Add spec for CALLER opcode

ed255 opened the pull request:
I have followed the pattern from https://github.com/appliedzkp/zkevm-specs/pull/83 but using the CallContext in the rw_table instead of the Block table for the value lookup.

Related issue: https://github.com/appliedzkp/zkevm-specs/issues/86

---

I've added two questions for doubts I had:
```
    # QUESTION: Where is it checked that address raw data is 160 bit length (20
    # bytes)?
    # QUESTION: Which is better of the two options?
    # a. rlc_decode(stack_value) == call_context_lookup()
    # b. stack_value == rlc_encode(call_context_lookup())
```

han0110 commented:
> Where is it checked that address raw data is 160 bit length (20 bytes)

The address from call context will always be verified by the step that triggers the call (e.g. `BeginTx`, `*CALL*`, `CREATE*`), and because `rw_table` guarantee read/write consistency, so we don't need to double-check if it's a valid address.

> Which is better of the two options?

Option b. sounds better IMHO, since we read the call context first, and does a base conversion (from `256` to `r`), and push to stack. But both are working for me actually.

CPerezz commented on specs/opcode/33CALLER.md:
```diff
@@ -0,0 +1,40 @@
+   - `address` is on top of stack
+   - `address` is in the rw table {call context, call ID, caller}
+4. Others:
+   - `address` is 20 bytes length
```
I would mention as @han0110 said that this is already constrained by the BeginTx gadget before.

So we're covered.

CPerezz reviewed (approved):
LGTM modulo applying @han0110 's replies.

han0110 reviewed (approved):
LGTM!

ed255 commented:
> Option b. sounds better IMHO, since we read the call context first, and do a base conversion (from `256` to `r`), then push to stack. But both work for me.

I've updated the spec to follow option b, and removed the notes about checking the address byte length.
